Challenge-4/Bot_2/src/agent.spec.ts

```diff
@@ -0,0 +1,459 @@
+import {
+  ethers,
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+} from "forta-agent";
+import { provideInitialize, provideHandleTransaction } from "./agent";
+import { createAddress } from "forta-agent-tools";
+import Helper from "./helper";
+import { getAlerts } from "forta-agent";
+
+jest.mock("forta-agent", () => ({
+  ...jest.requireActual("forta-agent"),
+  getAlerts: jest.fn(),
+}));
+
+describe("Compound test suite for lending and borrowing", () => {
+  let mockProvider = new MockEthersProvider();
+  let handleTransaction: HandleTransaction;
+  let txEvent: TestTransactionEvent;
+  let initialize: any;
+  let helper: Helper;
+  let usdcTokenAddress = "0xc3d688B66703497DAA19211EEdff47f25384cdc3";
+  let mockConfiguratorProxy = "0x316f9708bb98af7da9c68c1c3b5e79039cd336e3";
+  const functions = [
+    `function getConfiguration(address cometProxy) view returns (tuple(address governor, address pauseGuardian, address baseToken, address baseTokenPriceFeed, address extensionDelegate, uint64 supplyKink, uint64 supplyPerYearInterestRateSlopeLow, uint64 supplyPerYearInterestRateSlopeHigh, uint64 supplyPerYearInterestRateBase, uint64 borrowKink, uint64 borrowPerYearInterestRateSlopeLow, uint64 borrowPerYearInterestRateSlopeHigh, uint64 borrowPerYearInterestRateBase, uint64 storeFrontPriceFactor, uint64 trackingIndexScale, uint64 baseTrackingSupplySpeed, uint64 baseTrackingBorrowSpeed, uint104 baseMinForRewards, uint104 baseBorrowMin, uint104 targetReserves, tuple(address asset, uint8 decimals, uint256 conversionFactor)[] assetConfigs) configuration)`,
+    `function getUtilization() public view returns (uint)`,
+    `function getSupplyRate(uint utilization) public view returns (uint64)`,
+    `function getBorrowRate(uint utilization) public view returns (uint64)`,
+  ];
+  const OTHER_FUNCTION = "function otherFunction(address asset, uint amount)";
+  const Iface = new ethers.utils.Interface(functions);
+  const provideInterface = new ethers.utils.Interface([
+    "function supply(address asset, uint amount)",
+    "function withdraw(address asset, uint amount)",
+    OTHER_FUNCTION,
+  ]);
+
+  const mockAlerts = (alertId: string, kink: any, utilization: any) => {
+    (getAlerts as jest.Mock).mockResolvedValue({
+      alerts: [
+        {
+          alertId: alertId,
+          hasAddress: jest.fn().mockReturnValue(false),
+          metadata: {
+            kink: kink,
+            utilization: utilization,
+          },
+        },
+      ],
+      pageInfo: { hasNextPage: false },
+    });
+  };
+
+  beforeEach(() => {
+    handleTransaction = provideHandleTransaction();
+    initialize = provideInitialize(mockProvider as any);
+    helper = new Helper(mockProvider as any);
+    txEvent = new TestTransactionEvent().setBlock(0);
+  });
+
+  let setupMockProvider = async (
+    supplyKink: ethers.BigNumber,
+    borrowKink: ethers.BigNumber,
+    utilization: ethers.BigNumber,
+  ) => {
+    mockProvider.setNetwork(1);
```

Comment:
This could be done in the `beforeEach()`, since you shouldn't really need to call `setNetwork` every time `setupMockProvider` is called.